subsys/usb/class/mass_storage.c

```diff
@@ -640,6 +640,7 @@ static void mass_storage_bulk_out(u8_t ep,
 		usb_ep_set_stall(ep);
 		csw.Status = CSW_ERROR;
 		sendCSW();
+		usb_ep_clear_stall(ep);
```

Comment:
The MSC driver should not clear stall by itself, it's just wrong here. For MSC, the host uses the default pipe to clear a stall on the bulk endpoints (REQ_CLEAR_FEATURE) or perform a Reset Recovery (MSC_REQUEST_RESET + REQ_CLEAR_FEATURE for bulk OUT/IN endpoints).

I suggest to investigate the reason for the stall. For example, you can change line 639 to SYS_LOG_ERR and adjust the log level accordingly to get the MSC Stage.

Reply:
Thanks for inputs. I did debugging following are the observation,

1. After sendCsw() function invoked in mass_storage_bulk_in callback,stage is set to WAIT_CSW.

2. After step 1, mass_storage_bulk_in callback should be called. Instead mass_storage_bulk_out callback gets inovked. since stage is set to WAIT_CSW, default case is called which sets the stall bit.

3.  As u mentioned host initiates reset which invokes mass_storage_status_cb callback. 

As a solution i am thinking to clear stall bit mass_storage_status_cb callback when host initiates reset. Would this be acceptable solution.

Reply:
@jfischer-phytec-iot  any update on my comment

Reply:
@SavinayDharmappa Can you describe your setup in more detail? Can I reproduce it with Arduino 101?

Reply:
@jfischer-phytec-iot  It is not reproducible in Arduino_101.I have checked it. The once difference is we are using usb 2.0 where as arduino 101 usb is 1.0

Reply:
- What is the endpoint configuration? Does it come to the error with the sample ``sample/subsys/usb/mass`` ?
- Which version of Windows is it? Does the error also occur with other host systems?

Reply:
The usb endpoint configuration is same . The only difference is in DCFG control register we are setting to 1. Which means selecting usb 2.0 phy clock is from 30Mhz to 60Mhz.

When device is connected to linux system stalling is observed , but after multiple reset from host device recovers and gets detected as mass storage device . Note when linux system is host clearing of stalling bit is not neccessary from device side

Which version of Windows is it?
>>Windows 10

Does it come to the error with the sample sample/subsys/usb/mass ?
>> yes

Reply:
@jfischer-phytec-iot  I would further debug the issue and will try to find is there any another issue  causing problem in detecting device as mass storage on  windows. Mean while i want another patch to get merged so i will be removing patch which was fixing windows issue.